scripts/firedrake-install

```diff
@@ -790,6 +663,33 @@ def pip_uninstall(package):
             raise InstallError("pip claims it uninstalled %s more than 10 times.  Something is probably broken.", package)
 
 
+def build_and_install_petsc():
+    import hashlib
+    tarball = "eigen-3.3.3.tgz"
+    url = "http://bitbucket.org/eigen/eigen/get/3.3.3.tar.gz"
+    sha = hashlib.sha256()
+    expect = "94878cbfa27b0d0fbc64c00d4aafa137f678d5315ae62ba4aecddbd4269ae75f"
+    if not os.path.exists(tarball):
+        log.info("Downloading Eigen from '%s' to '%s'" % (url, tarball))
+        check_call(["curl", "-L", "-o", tarball, url])
+    else:
+        log.info("Eigen tarball already downloaded to '%s'" % tarball)
+    log.info("Checking Eigen tarball integrity")
+    with open(tarball, "rb") as f:
+        while True:
+            data = f.read(65536)
+            if not data:
+                break
+            sha.update(data)
```

Comment:
There ought to be a more concise way to say this loop, but perhaps not as memory efficient?